lib/roast/workflow/file_state_repository.rb

```diff
@@ -90,9 +92,9 @@ def find_step_files(session_dir)
       end
 
       def find_step_before(step_files, target_step_name)
-        # First try to find if we have the exact previous step
         step_files.each_with_index do |file, index|
-          next unless file.end_with?("_#{target_step_name}.json")
+          state_data = load_state_file(file)
+          next if state_data[:step_name] != target_step_name && state_data[:step_name] != target_step_name.to_s
```

Comment:
```suggestion
          next if state_data[:step_name] != target_step_name.to_s
```
Do we need both checks here? In what case will `target_step_name` not be a String? If there is a case like that, why would we want to compare it to `state_data[:step_name]` 🤔